manifests/vhost/custom.pp

```diff
@@ -0,0 +1,38 @@
+# See README.md for usage information
+define apache::vhost::custom(
+  $content,
+  $ensure = 'present',
+  $priority = '25',
+) {
+  include ::apache
+
+  ## Apache include does not always work with spaces in the filename
+  $filename = regsubst($name, ' ', '_', 'G')
+
+  ::apache::custom_config { $filename:
+    ensure   => $ensure,
+    confdir  => $::apache::vhost_dir,
+    content  => $content,
+    priority => $priority,
+  }
+
+  # NOTE(pabelanger): This code is duplicated in ::apache::vhost and needs to
+  # converted into something generic.
+  if $::apache::vhost_enable_dir {
+    $vhost_symlink_ensure = $ensure ? {
+      present => link,
+      default => $ensure,
+    }
+
+    file { "${priority}-${filename}.conf symlink":
+      ensure  => $vhost_symlink_ensure,
+      path    => "${::apache::vhost_enable_dir}/${priority}-${filename}.conf",
+      target  => "${::apache::vhost_dir}/${priority}-${filename}.conf",
+      owner   => 'root',
+      group   => $::apache::params::root_group,
```

Comment:
So, honestly I don't know the answer.  For the most part, this is copied verbatim from apache::vhost, so I would say the issue is there too.

Moving forward, I think this code should be broken out from both apache::vhost and apache::vhost::custom into a shared class.  Making bug fixes there too.

I didn't want to do that here, as to not make things more complicated.